### PR TITLE
Replace 'multi_dep' loader with new 'from_deps' transforms

### PR DESCRIPTION
**Pull request by ahal**
### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [x] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [x] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Comment by ahal** [comment]
This work was initially part of https://github.com/mozilla/application-services/pull/5619 but I decided to split it out into a dedicated PR to make it easier to review.

While a lot is changing here, running `taskgraph full -J --diff -p taskcluster/test/params` results in an identical graph (other than a new attribute being added to beetmover and signing tasks). So I'm confident that it is correct.

**Review by jcristau — COMMENTED** [review]
Very nice to see the loader go. :)

**Review by bhearsum — DISMISSED** [review]
Excellent improvement!

File: taskcluster/app_services_taskgraph/transforms/__init__.py
Comment (bendk) [reply taskcluster/app_services_taskgraph/transforms/__init__.py:49]: Single quotes over double quotes triggers me, but it's worth it for the consistency :)  I made https://github.com/mozilla/application-services/issues/5628 for this.

**Review by bendk — APPROVED** [review]
Looks good to me.  I'm thinking we should wait until after the 115 release to merge, just in case there's any unexpected errors.  Does that sound good to others?

**Comment by ahal** [comment]
> Looks good to me. I'm thinking we should wait until after the 115 release to merge, just in case there's any unexpected errors. Does that sound good to others?

This sounds good to me, though I did just successfully test the beetmover tasks, so there's a small chance they could be landed today. Landing this blocks that, so if we hold off it means for sure no beetmovers. That's fine by me, just calling it out.

**Review by bendk — APPROVED** [review]

**Comment by bendk** [comment]
The 115 release went out yesterday, so this is clear to land in main now.

**Pull request merged**
